lib/matplotlib/dates.py

```diff
@@ -1905,6 +1911,49 @@ def axisinfo(self, unit, axis):
                               default_limits=(datemin, datemax))
 
 
-units.registry[np.datetime64] = DateConverter()
-units.registry[datetime.date] = DateConverter()
-units.registry[datetime.datetime] = DateConverter()
+# The following is so that we can set the converter for dates
```

Comment:
Perhaps (not sure) a helper class which is not instantiated but just serves as namespace to hold the globals? e.g. something along the lines of
```python
class _RcHelper:
    """
    Never instantiated, only used to manage rcParams.
    """
    conv_st = ...
    int_mult = ...

    @classmethod
    def set_converter(cls, s):
        cls.conv_st = s
        cls.register_converters()

    @classmethod
    def register_converters(cls):
        if cls.conv_st == "concise": ...


def _validate_date_converter(s):
    ...
        mdates._RcHelper.set_converter(s)
```

Reply:
Done!